### PR TITLE
Updates and fixes for the pyside 3P package on Linux

### PR DESCRIPTION
**Pull request by spham-amzn**
* Update / Fix the git URL
* Update the dependent packages for linux
* Update the patchelf to use `--force-rpath` to set the rpath properly
* Add a symlink to the python package in the docker build script to correct the build

### CODE REVIEW

**Comment by amzn-changml** [comment]
Let me know if this is ready to promote to dev for AR use

File: package-system/pyside2/Dockerfile [hunk 1]
```diff
@@ -9,14 +9,27 @@
...
 #
 
-FROM public.ecr.aws/ubuntu/ubuntu:20.04_stable
- 
-WORKDIR /data/workspace
+# The cpu architecture to base the docker base script from
+ARG INPUT_ARCHITECTURE=amd64
+
+# The root to base the docker script base from
+ARG INPUT_IMAGE=ubuntu:20.04
```
[the remark this answers is not in the document]
Comment (spham-amzn) [reply package-system/pyside2/Dockerfile:16]: The last time I tried, ECR didn't have the aarch64 based container that followed the pattern that I used before, and I started to move away from any AWS specific dependencies

File: package_build_list_host_linux.json [hunk 2]
```diff
@@ -20,7 +20,8 @@
         "libsamplerate-0.2.1-rev2-linux": "Scripts/extras/pull_and_build_from_git.py ../../package-system/libsamplerate --platform-name Linux --package-root ../../package-system --clean",
         "mcpp-2.7.2_az.1-rev1-linux": "package-system/mcpp/get_and_build_mcpp.py mcpp-2.7.2_az.1-rev1",
         "OpenMesh-8.1-rev3-linux": "Scripts/extras/pull_and_build_from_git.py ../../package-system/OpenMesh --platform-name Linux --package-root ../../package-system --clean",
-        "pyside2-5.15.2.1-py3.10-rev4-linux": "Scripts/extras/pull_and_build_from_git.py ../../package-system/pyside2 --platform-name Linux --package-root ../../package-system/pyside2/temp --clean",
+        "pyside2-5.15.2.1-py3.10-rev7-linux": "Scripts/extras/pull_and_build_from_git.py ../../package-system/pyside2 --platform-name Linux --package-root ../../package-system/pyside2/temp --clean",
+        "pyside2-5.15.2.1-py3.10-rev7-linux-aarch64": "Scripts/extras/pull_and_build_from_git.py ../../package-system/pyside2 --platform-name Linux-aarch64 --package-root ../../package-system/pyside2/temp --clean",
```
[the remark this answers is not in the document]
Comment (spham-amzn) [reply package_build_list_host_linux.json:24]: The GHA to invoke the build was searching for the aarch64 key from the linux-x86 host file, not the aarch one. This was done to satisfy that. I was still able to run from a graviton EC2 with the aarch host list anyways,

**Comment by spham-amzn** [comment]
> Let me know if this is ready to promote to dev for AR use

What is the package URL to use to test these artifacts?

File: Scripts/extras/package_downloader.py [hunk 3]
```diff
@@ -20,7 +20,7 @@
 
 
 # used if LY_PACKAGE_SERVER_URLS is not set.
-DEFAULT_LY_PACKAGE_SERVER_URLS = "https://d2c171ws20a1rv.cloudfront.net"
+DEFAULT_LY_PACKAGE_SERVER_URLS = "https://d2c171ws20a1rv.cloudfront.net;https://d3t6xeg4fgfoum.cloudfront.net"
```
[the remark this answers is not in the document]
Comment (spham-amzn) [reply Scripts/extras/package_downloader.py:23]: If its no longer active I can get rid of it in the PR

File: package-system/pyside2/build_config.json [hunk 4]
```diff
@@ -1,8 +1,8 @@
 {
-    "git_url": "https://code.qt.io/cgit/pyside/pyside-setup.git",
+    "git_url": "https://code.qt.io/pyside/pyside-setup",
     "git_tag": "v5.15.2.1",
     "package_name": "pyside2",
-    "package_version": "5.15.2.1-py3.10-rev4",
+    "package_version": "5.15.2.1-py3.10-rev7",
     "package_url": "https://code.qt.io/cgit/pyside/pyside-setup.git",
```
[the remark this answers is not in the document]
Comment (spham-amzn) [reply package-system/pyside2/build_config.json:6]: No, the git_url used for cloning in this case will respond with a 404 if you try to access it directly. The current package_url is still valid as it points to the pyside2 git super project landing page.

**Comment by amzn-changml** [comment]
> > Let me know if this is ready to promote to dev for AR use
> 
> What is the package URL to use to test these artifacts?

https://d1gg6ket0m44ly.cloudfront.net would be the dev 3p URL

File: Scripts/extras/package_downloader.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (amzn-changml) [reply Scripts/extras/package_downloader.py:23]: It's still in use for past versions, but if the pyside and python changes move these dependencies to the prod 3p bucket, then we don't need it going forward.

**Review by lemonade-dm — APPROVED** [review]

**Review by amzn-changml — APPROVED** [review]

**Pull request merged**